src/OpenTelemetry.Extensions/Logs/LogToActivityEventConversionOptions.cs

```diff
@@ -34,4 +34,17 @@ public class LogToActivityEventConversionOptions
     /// Gets or sets the callback action used to convert log scopes into <see cref="ActivityEvent"/> tags.
     /// </summary>
     public Action<ActivityTagsCollection, int, LogRecordScope> ScopeConverter { get; set; } = DefaultLogStateConverter.ConvertScope;
+
+    /// <summary>
+    /// Gets or sets the callback method allowing to filter out particular <see cref="ActivityEvent"/>.
+    /// </summary>
+    /// <remarks>
+    /// The filter callback receives the <see cref="ActivityEvent"/> for the
+    /// processed logRecord and should return a boolean.
+    /// <list type="bullet">
+    /// <item>If filter returns <see langword="true"/> the event is collected.</item>
+    /// <item>If filter returns <see langword="false"/> the event is filtered out (NOT collected).</item>
+    /// </list>
+    /// </remarks>
+    public Func<ActivityEvent, bool>? Filter { get; set; }
```

Comment:
Thanks for this PR @trejjam! Everything seems fine to me but I have one comment about the design. Performing the filter on `ActivityEvent` means a lot of processing and allocations have already occurred by the time the filter is called. It would be more efficient to filter on the `LogRecord` before the `ActivityEvent` is ever created. Would that meet your needs or the `LogRecord` is too difficult to work with given the sort of strange scope/state mechanics?

Reply:
Using `LogRecord` is fine. The workflow is changed